### PR TITLE
AudioManager

### PR DESCRIPTION
**Pull request by dminlee02**
Created AudioManager and music_events to play music. I have the basic functionality of playing different (for now 10 seconds, looped) songs for the build and dungeon phases.
For AudioData, I might add future variables for volume, looped, etc. so we can have more fine-grain control for both the music and sound effects.

### CODE REVIEW

File: src/events.h [hunk 1]
```diff
@@ -60,6 +59,13 @@ namespace events {
     };
     extern signal<void(StickData &)> stick_event;
 
+    // Audio
+    struct AudioData {
+        std::string filename;
+    };
```
Comment (AznCelebration) [inline src/events.h:65]: Are u planning to have other variables in this struct? If not, then just the string would be fine

File: src/audio/audio_manager.cpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+AudioManager::AudioManager() {
+    events::music_event.connect([&](const events::AudioData &d) {
+        // Stop the current music from playing, if any
+        if (music.getStatus() == sf::SoundSource::Status::Playing) {
+            music.stop();
+        }
+        
+        std::string filename = d.filename;
+        if (!music.openFromFile(filename))
+        {
```
Comment (richard1) [inline src/audio/audio_manager.cpp]: Curly brace on previous line pls

File: src/audio/audio_manager.cpp [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            fprintf(stderr, "AudioManager: Cannot open %s\n", filename);
```
Comment (richard1) [inline src/audio/audio_manager.cpp]: Can use `std::cerr` instead for consistency

**Review by richard1 — COMMENTED** [review]
Minor comments, but looks good overall.  Good job!

**Review by Pukami — APPROVED** [review]
Worked fine on my computer, and I don't see any problems.

File: src/events.h (hunk 1 above)
Comment (jytang) [reply src/events.h:65]: or a typedef, either way

Comment (jytang) [reply src/events.h:65]: Wait, he said this in the pr description: "For AudioData, I might add future variables for volume, looped, etc. so we can have more fine-grain control for both the music and sound effects." So it's fine.

**Review by jytang — APPROVED** [review]
Looks good. For the future, as for sound effects and such like (latency sensitive), we may want to load audio assets into memory at startup.

File: src/audio/audio_manager.cpp (hunk 3 above)
Comment (metakirby5) [reply src/audio/audio_manager.cpp]: This also does not compile with `clang++`, you need to use `filename.c_str()` if you want to use fprintf. Or just use `std::cerr`.

**Review by metakirby5 — CHANGES_REQUESTED** [review]
Please add the following line to the Makefile, for us non-VS users:

```make
LIBS            += -lsfml-system -lsfml-audio
```

Other than that and the fprintf thing, looks good and works on my machine.
 🚢

**Review by richard1 — APPROVED** [review]
I added a commit (which adds a space).  LGTM

**Pull request merged**